### PR TITLE
convert version numbers to stricly numerical in Siesta easyblock

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyblocks/s/siesta.py [hunk 1]
```diff
@@ -125,7 +126,7 @@ def configure_step(self):
         # Populate start_dir with makefiles
         run_cmd(os.path.join(start_dir, 'Src', 'obj_setup.sh'), log_all=True, simple=True, log_output=True)
 
-        if loose_ver < LooseVersion('4.1-b2'):
+        if loose_ver < LooseVersion('4.1.0.2'):
```
Comment (migueldiascosta) [inline easybuild/easyblocks/s/siesta.py:129]: shouldn't this be `4.1.0.22` then? (in case someone creates an easyconfig for `4.1-b1` -> `4.1.0.21`)

**Comment by migueldiascosta** [comment]
I suppose this will also solve https://github.com/easybuilders/easybuild-easyconfigs/issues/13559

File: easybuild/easyblocks/s/siesta.py (hunk 1 above)
Comment (akesandgren) [reply easybuild/easyblocks/s/siesta.py]: Yeah it should, so I adjusted things a bit...

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1-b4-foss-2018b.eb_ 
 * **FAIL (build issue)** _Siesta-4.1-MaX-1.0-intel-2019b.eb_ (partial log available at https://gist.github.com/df62007d6016f71a77c9688b75d63668)
 * **FAIL (build issue)** _Siesta-4.1-MaX-1.0-intel-2019b-PEXSI.eb_ (partial log available at https://gist.github.com/047115d4007bf354c4f320cc51cf3fa8)
 * **SUCCESS** _foss-2017b.eb_ 
 * **SUCCESS** _Siesta-4.0-foss-2017b.eb_ 
 * **SUCCESS** _icc-2017.1.132-GCC-6.3.0-2.27.eb_ 
 * **FAIL (build issue)** _icc-2018.3.222-GCC-7.3.0-2.30.eb_ (partial log available at https://gist.github.com/7c7b5924f4a73a847245a9220df9d662)
 * **SUCCESS** _ifort-2017.1.132-GCC-6.3.0-2.27.eb_ 
 * **FAIL (build issue)** _ifort-2018.3.222-GCC-7.3.0-2.30.eb_ (partial log available at https://gist.github.com/3226f6c5a240b589fe2f852105b24e46)
 * **SUCCESS** _intel-2017a.eb_ 
 * **FAIL (build issue)** _intel-2018b.eb_ (partial log available at https://gist.github.com/a1ab9f535d60ea0e0e5cffd21be4211d)
 * **SUCCESS** _Siesta-4.0.1-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.0-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.1-b2-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.1-b3-intel-2017a.eb_ 
 * **FAIL (build issue)** _Siesta-4.1-b4-intel-2018b.eb_ (partial log available at https://gist.github.com/4afc9a2c6c1140811ad77b059cbd2505)

Build succeeded for 10 out of 16 (9 easyconfigs in total)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/d5843471d916b48844bdef1cac5e4272 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **FAIL (unhandled exception: '<' not supported between instances of 'str' and 'int')Traceback (most recent call last):
  File "/p/project/cswmanage/jusuf/stages/Upstream/software/EasyBuild/20210904-dev/lib/python3.6/site-packages/easybuild/main.py", line 118, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/p/project/cswmanage/jusuf/stages/Upstream/software/EasyBuild/20210904-dev/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3713, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/p/project/cswmanage/jusuf/stages/Upstream/software/EasyBuild/20210904-dev/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3601, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/p/project/cswmanage/jusuf/stages/Upstream/software/EasyBuild/20210904-dev/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3454, in run_step
    step_method(self)()
  File "/tmp/eb-qiu1zbp7/included-easyblocks-f7nse690/easybuild/easyblocks/siesta.py", line 129, in configure_step
    if loose_ver < LooseVersion('4.1.0.2.2'):
  File "/usr/lib64/python3.6/distutils/version.py", line 52, in __lt__
    c = self._cmp(other)
  File "/usr/lib64/python3.6/distutils/version.py", line 337, in _cmp
    if self.version < other.version:
TypeError: '<' not supported between instances of 'str' and 'int'
** _Siesta-4.1-MaX-1.0-intel-2019b-PEXSI.eb_ 

Build succeeded for 0 out of 1 (1 easyconfigs in total)
jsfl03.jusuf - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/0ee75adf9d8f32e730f61624f3d61962 for a full test report.

**Comment by SebastianAchilles** [comment]
Fails for `Siesta-4.1-MaX-1.0-intel-2019b-PEXSI.eb` with

```
  File "/tmp/eb-qiu1zbp7/included-easyblocks-f7nse690/easybuild/easyblocks/siesta.py", line 129, in configure_step
    if loose_ver < LooseVersion('4.1.0.2.2'):
  File "/usr/lib64/python3.6/distutils/version.py", line 52, in __lt__
    c = self._cmp(other)
  File "/usr/lib64/python3.6/distutils/version.py", line 337, in _cmp
    if self.version < other.version:
TypeError: '<' not supported between instances of 'str' and 'int
```

**Comment by smoors** [comment]
Test report by @smoors
#### Overview of tested easyconfigs (in order)

Build succeeded for 0 out of 0 (1 easyconfigs in total)
node301.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/dfca41975cf83b02f8ee50c3227a8e80 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node301.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/6a4a4cb766cc13e1f93b529b9bdeeb9c for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1-b4-foss-2018b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node308.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/560cbf452ce5b0a92c87acbf3ff5f241 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1-5-intel-2019b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node357.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/eccba20391b8437620cf5412f60bb8e5 for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @akesandgren!

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.0.1-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.0-foss-2017b.eb_ 
 * **SUCCESS** _Siesta-4.0-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.1-b2-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.1-b3-intel-2017a.eb_ 
 * **SUCCESS** _Siesta-4.1-b4-foss-2018b.eb_ 
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b.eb_ 
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b-PEXSI.eb_ 
 * **FAIL (build issue)** _icc-2018.3.222-GCC-7.3.0-2.30.eb_ (partial log available at https://gist.github.com/098a0dc8d71e4fd2b1a48e3958226028)
 * **FAIL (build issue)** _ifort-2018.3.222-GCC-7.3.0-2.30.eb_ (partial log available at https://gist.github.com/7f815ad0f79a1dcb8857394b22b25816)
 * **FAIL (build issue)** _intel-2018b.eb_ (partial log available at https://gist.github.com/2f4b9352faa8b4aa2a8dae99e4b697fa)
 * **FAIL (build issue)** _Siesta-4.1-b4-intel-2018b.eb_ (partial log available at https://gist.github.com/baf796372a7c037bfad5d1339db26617)

Build succeeded for 8 out of 12 (9 easyconfigs in total)
c2dhpc - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/cf5b713447d7b856d35b301cf2f9c67c for a full test report.

**Comment by migueldiascosta** [comment]
the failures above are only because of the license for `intel/2018`, but `Siesta-4.1-b4` builds with `foss/2018` and other versions of `Siesta` build with the versions of `intel` that are already installed, so there shouldn't be any problem

**Pull request merged**